pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -189,6 +196,235 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+// registryIsMatch returns true if flags in the registry entry indicate that
+// its URL might not be used as-is, but will have URLs matched against it in
+// order to selectively apply configuration flags to an unlisted URL.
+func registryIsMatch(reg Registry) bool {
+	return reg.Insecure || reg.Blocked
```

Comment:
AFAICT this is essentially an “ignore parsing errors for this entry” predicate; that’s just not viable in the v2 model, where there is a single registry object with flags.  A registry being marked as `Insecure` does not at all mean that the `Prefix` or `URL` field won’t be used for pulls (see how `postProcessRegistries` enforces the `Insecure`/`Blocked` values to be consistent across all instances with the same `URL`.)

Unless I am missing something, this essentially removes any guarantees that the `URL`/`Prefix` are well-formed.